src/tftp/server.rs

```diff
@@ -95,24 +96,30 @@ fn handle_client(socket: UdpSocket, mut server: TFTPServer, client_addr: SocketA
         }
 
         let mut buf = [0 as u8; 1024];
-        let (count, addr) = socket
-            .recv_from(&mut buf)
-            .expect("Failed to read socket fd");
-        let raw_msg = &buf[..count];
-
-        if addr != client_addr {
-            let error_packet = ErrorPacket::new(TFTPError::UnknownTID);
-            socket.send_to(&error_packet.serialize(), addr).unwrap();
-        }
+        match socket.recv_from(&mut buf){
+            Ok((count, addr)) => {
+                let raw_msg = &buf[..count];
+
+                if addr != client_addr {
+                    let error_packet = ErrorPacket::new(TFTPError::UnknownTID);
+                    socket.send_to(&error_packet.serialize(), addr).unwrap();
+                }
 
-        server.run(raw_msg);
+                server.run(raw_msg);
+            },
+            Err(e) => {
+                eprintln!("Client connection error: {}", e);
+                break;
+            }
+        }
     }
-    // });
 }
 
 pub fn handle_new_client(client_addr: SocketAddr, rq_packet: &[u8]) {
+    const sock_dur: Option<Duration> = Some(Duration::from_secs(5));
```

Comment:
`sock_dur` isn't a function constant (as it's a program property), so it's best placed at the top of the file